test/assert.go

```diff
@@ -383,9 +383,10 @@ func (assert *Assert) fuzzer(fuzzer filler, circuit, w frontend.Circuit, b backe
 	// fuzz a witness
 	fuzzer(w, curve)
 
-	err := IsSolved(circuit, w, curve, b)
+	errVars := IsSolved(circuit, w, curve, WithBackend(b))
+	errConsts := IsSolved(circuit, w, curve, WithBackend(b), SetAllVariablesAsConstants())
 
-	if err == nil {
+	if errVars == nil && errConsts == nil {
```

Comment:
@ivokub what useful information can we (systematically) deduce when `(errVars == nil) != (errConsts == nil)` ?

Reply:
As an example, the cause for my bug was https://github.com/ConsenSys/gnark/blob/master/std/math/bits/conversion_binary.go#L68. Here the bit decomposition for constants and variables is slightly different. For variables the decomposed bits have to add up to the variable value, but not for constants.

I wouldn't say the behavior is wrong for constants, but it definitely is unexpected.